### PR TITLE
Remove GH App linked accounts implementation

### PR DESCRIPTION
**Pull request by pjlast**
This PR removes the GitHub App Linked Accounts implementation. This was necessary when we were heading towards using both repo- and user-based permissions syncing, in order to ensure that repo-based permissions syncing is not giving access to repositories that the user-account token should not have access to.

However, since we want to solely use user-tokens for permissions syncing in the near future, this is no longer necessary and adds unnecessary complexity and confusion.

## Test plan

All tests still pass and tested locally.

### CODE REVIEW

File: internal/database/external_accounts.go [hunk 1]
```diff
@@ -384,54 +381,6 @@ type ExternalAccountsListOptions struct {
 	*LimitOffset
 }
 
-func (s *userExternalAccountsStore) UpdateGitHubAppInstallations(ctx context.Context, acct *extsvc.Account, installations []gh.Installation) error {
-	acctInstallations, err := s.List(ctx, ExternalAccountsListOptions{
-		ServiceType:    extsvc.TypeGitHubApp,
-		AccountIDLike:  fmt.Sprintf("%%/%s", acct.AccountID),
```
Comment (mrnugget) [inline internal/database/external_accounts.go:390]: I think the code you removed was the only user of the `AccountIDLike` option. I think we can also remove it?

**Review by mrnugget — APPROVED** [review]

File: internal/database/external_accounts.go (hunk 1 above)
Comment (pjlast) [reply internal/database/external_accounts.go:390]: You are correct

**Comment by unknwon** [comment]
> However, since we want to solely use user-tokens for permissions syncing in the near future, this is no longer necessary and adds unnecessary complexity and confusion.

Kindly asking for a doc link since I no longer actively monitoring IAM team channel 😁

**Comment by mrnugget** [comment]
> Kindly asking for a doc link since I no longer actively monitoring IAM team channel 😁

There is no doc link, it just came up in multiple meetings and when talking about scale testing.

**Comment by unknwon** [comment]
> > Kindly asking for a doc link since I no longer actively monitoring IAM team channel 😁
> 
> There is no doc link, it just came up in multiple meetings and when talking about scale testing.

Thanks for the context! Curious what was the "takeaway" that triggered this removal?

**Comment by pjlast** [comment]
@unknwon for larger clients we have to focus on using user tokens to sync permissions, because of the number of requests that we need to make to stay up to date. Then there is also the fact that this solution would only solve this edge case for cases where people use GitHub App, but there is no desire to make GitHub App the only way to interface with GitHub. So if PATs will still stick around, we'd rather remove the code that causes confusion for solving a single edge case for 1 specific setup, and focus on solving the problem across the board.

**Comment by unknwon** [comment]
> @unknwon for larger clients we have to focus on using user tokens to sync permissions, because of the number of requests that we need to make to stay up to date. Then there is also the fact that this solution would only solve this edge case for cases where people use GitHub App, but there is no desire to make GitHub App the only way to interface with GitHub. So if PATs will still stick around, we'd rather remove the code that causes confusion for solving a single edge case for 1 specific setup, and focus on solving the problem across the board.

Thank you, makes sense!

**Pull request merged**